src/client/Client.cc

```diff
@@ -3091,7 +3091,19 @@ void Client::handle_mds_map(const MConstRef<MMDSMap>& m)
 	continue;
       }
       if (newstate >= MDSMap::STATE_ACTIVE) {
-	if (oldstate < MDSMap::STATE_ACTIVE) {
+	if (oldstate <= MDSMap::STATE_ACTIVE) {
```

Comment:
`STATE_ACTIVE` is `13` while `STATE_STOPPING` is `14`. So we're skipping `STATE_STOPPING` it seems.

Reply:
also if we're concerning with the stopping/stopped states then why not check them explicitly? and as greg mentioned in one of the comments - why flush the caps when there is no guarantee of them being actually flushed, instead how about creating a queue of the requests and sending them to the MDS when the MDS failover takes place? so that clients can continue with the data ops while they can survive with the cached metadata till the time mds comes alive, makes sense?

Reply:
Actually this should be in recover case and the corresponding MDS is just recovered and set the state to ACTIVE. While from the mds code the `STOPPING` do not means it won't work and won't accept the client requests.

I think I should switch `if (oldstate <= MDSMap::STATE_ACTIVE) {` back to `if (oldstate < MDSMap::STATE_ACTIVE) {`.

Reply:
I think the `flushing_cap_tids` is doing this. Let me check it again later. Thanks @dparmar18